src/servers/http_server_v2.cc

```diff
@@ -715,25 +715,39 @@ WriteDataToJsonHelper(
     const rapidjson::Value& shape, int shape_index, T* base, int* counter,
     const DataType dtype)
 {
-  for (int i = 0; i < shape[shape_index].GetInt(); i++) {
-    if ((shape_index + 1) != (int)shape.Size()) {
-      rapidjson::Value response_output_array(rapidjson::kArrayType);
-      WriteDataToJsonHelper(
-          &response_output_array, allocator, shape, shape_index + 1, base,
-          counter, dtype);
-      response_output_val->PushBack(response_output_array, allocator);
+  if (shape_index == (int)shape.Size()) {
+    if (dtype != TYPE_STRING) {
+      rapidjson::Value data_val((T)(base[*counter]));
```

Comment:
base is already a T*, cast (T) is unnecessary.

Reply:
Correct. Will fix in a follow up PR